src/_zkapauthorizer/_plugin.py

```diff
@@ -205,6 +207,7 @@ def get_downloader(cap):
             node_config,
             store=self._get_store(node_config),
             get_downloader=get_downloader,
+            ##            get_uploader=make_fail_uploader(work_in_progress_error),
```

Comment:
I'm not sure the web resource will need the uploader at all.  The "turn on recovery" already gets the `setup_replication` callable.  If upload is necessary, maybe it will be hidden behind that function (but also that function currently only writes out some config, does not directly do any uploads itself).